src/lib/rest/rest.cpp

```diff
@@ -1072,6 +1101,18 @@ static int connectionTreat
     LM_T(LmtUriParams, ("'default' value for notifyFormat (ciP->outFormat == %d)): '%s'", ciP->outFormat, ciP->uriParam[URI_PARAM_NOTIFY_FORMAT].c_str()));
   }
 
+  //
+  // Here, if the incoming request was too big, return error abouyt it
+  //
+  if (ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE)
```

Comment:
What about if the client is not honoring the Content-Length header? For example, he uses `Content-Length: 50` but the paylod is 1.5 MB long. Maybe an extra check in the "copy chunk" logic to detect when the payload has grown too much (in fact, I'm not sure if with the current "copy chunk" logic the ciP->payload could be "overflow" in that situation...)?

(That case would be a rare situation, we don't need to cover this is this PR if you find it is hard. We can open an issue about it).